### ISSUE TITLE
Secrets with a plus sign not recognized

### ISSUE THREAD

**Issue by icelander** [mattermost/mattermost-plugin-jira#33]
# Summary

Testing the Jira plugin with a `curl` is a common way of figuring out if it's working. However, if there is a `+` in the secret it is not recognized by the plugin.

# Environment

* Mattermost Server: 5.8.0

# Steps to Reproduce

1. Edit `config.json` so that the `PluginSettings.Plugins.jira.secret` value has a `+` in it. For example: `KBcCZm_JWQWM2WD9tr5IMo+fUo5zGJwS`
2. `curl` the URL:

```bash
curl -XPOST "https://mattermost.example.com/plugins/jira/webhook?secret=KBcCZm_JWQWM2WD9tr5IMo+fUo5zGJwS&team=a-team&channel=town-square
```

# Expected Behavior

`curl` command outputs this:

```
EOF
```

# Observed Behavior

`curl` command outputs this:

```
You must provide the configured secret
```

# Recommended Fixes

Scrub secrets of `+` character when generating them.

**Comment by levb** [comment]
Right. We can (must?!)  URI-encode the secret.

BTW, your secret should be fine if you replace `+` with `%2B` in that URL

**Comment by jasonblais** [comment]
Resolved by https://github.com/mattermost/mattermost-plugin-jira/pull/53, to be included in Jira v2.0 build next month.
